javascript/javascript3/week3/shoppingCart.html

```diff
@@ -0,0 +1,130 @@
+                        const exchangeValue = (this.price * exchangeRate).toFixed(2) + currency;
+                        console.log(exchangeValue);
+                    })
+                    .catch((error) => {
+                        throw error;
+                    });
+            }
+        }
+
+        class ShoppingCart {
+            constructor() {
+                this.products = [];
+            }
+
+            addProduct(product) {
+                this.products.push(product);
+            }
+
+            removeProduct(product) {
+
+                const index = this.products.indexOf(product);
+                if (index > -1) {
+                    this.products.splice(index, 1);
+                }
+            }
+
+            searchProduct(productName) {
+                const index = this.products.indexOf(productName);
+                let searchedProducts = [];
+                if (index > -1) {
+                    searchedProducts.push(this.products[index]);
+                }
+                return (this.searchedProducts = searchedProducts);
+            }
+
+            getTotal() {
+                let totalAmount = 0;
+                this.products.forEach((product) => (totalAmount += product.price));
+                this.totalAmount = totalAmount;
+            }
+
+            renderProducts() {
+                this.products.map((product) => {
+
+                    console.log(product.name);
+                    console.log(product.price);
+
+
+
+
+                });
+            }
+
+
+
+            async getUser() {
+                try {
+                    const use = await fetch("https://jsonplaceholder.typicode.com/users/1");
+                    const response = await use.json();
+                    this.user = response;
+                    console.log(this.user.name);
+                } catch (error) {
+                    console.log(error);
+                }
+            }
+        }
+
+
+
+
+        const shoppingCart = new ShoppingCart();
+
+        const flatscreen = new Product("flat-screen", 5000);
+        shoppingCart.addProduct(flatscreen);
+        const smartTV = new Product("smart-tv", 2500);
+        shoppingCart.addProduct(smartTV);
+        const kitchenMachine = new Product("kitchen-machine", 10000);
+        shoppingCart.addProduct(kitchenMachine);
+        const laptop = new Product("laptop", 15000);
+        shoppingCart.addProduct(laptop);
+        const digitalPen = new Product("digital-pen", 100);
+        shoppingCart.addProduct(digitalPen);
+        digitalPen.convertToCurrency("EUR");
+        shoppingCart.removeProduct(flatscreen);
+        shoppingCart.searchProduct(kitchenMachine);
+
+        console.log(shoppingCart);
+
+        const allProducts = shoppingCart.products;
+
+        shoppingCart
+            .getUser()
+
+        .then(shoppingCart.renderProducts(allProducts))
+            .then(shoppingCart.getTotal())
+    </script>
```

Comment:
Please extract Javascript code into a separate file to make it clearer.

Reply:
@florinmaticu @aleks-hyf I've separated js from html. And implemented code with your suggestions. Thank you! :)